tests/versioning_test.go

```diff
@@ -397,6 +397,82 @@ func (s *versioningIntegSuite) dispatchNewWorkflowStartWorkerFirst() {
 	s.Equal("done!", out)
 }
 
+func (s *versioningIntegSuite) TestDisableLoadUserDataDefaultTasksBecomeUnversioned() {
+	dc := s.testCluster.host.dcClient
+	dc.OverrideValue(dynamicconfig.MatchingNumTaskqueueReadPartitions, 1)
+	defer dc.RemoveOverride(dynamicconfig.MatchingNumTaskqueueReadPartitions)
+	dc.OverrideValue(dynamicconfig.MatchingNumTaskqueueWritePartitions, 1)
+	defer dc.RemoveOverride(dynamicconfig.MatchingNumTaskqueueWritePartitions)
+
+	tq := s.randomizeStr(s.T().Name())
+	v0 := s.prefixed("v0")
+
+	// Register a versioned "v0" worker to execute a single workflow task to constrain a workflow on the task queue to a
+	// compatible set.
+	ch := make(chan struct{}, 1)
+	wf1 := func(ctx workflow.Context) (string, error) {
+		close(ch)
+		workflow.GetSignalChannel(ctx, "unblock").Receive(ctx, nil)
+		return "done!", nil
+	}
+
+	w1 := worker.New(s.sdkClient, tq, worker.Options{
+		BuildID:                          v0,
+		UseBuildIDForVersioning:          true,
+		MaxConcurrentWorkflowTaskPollers: numPollers,
+	})
+	w1.RegisterWorkflow(wf1)
+	s.NoError(w1.Start())
+	defer w1.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	s.addNewDefaultBuildId(ctx, tq, v0)
+	s.waitForPropagation(ctx, tq, v0)
+
+	// Start the first workflow while the task queue is still considered versioned.
+	// We want to verify that if a spooled task with a "compatible" versioning directive doesn't block a spooled task
+	// with a "default" directive.
+	// This should never happen in practice since we dispatch "default" tasks to the unversioned task queue but the test
+	// verifies this at a functional level.
+	run1, err := s.sdkClient.ExecuteWorkflow(ctx, sdkclient.StartWorkflowOptions{TaskQueue: tq}, wf1)
+	s.NoError(err)
+
+	// Wait for first WFT and stop the worker
+	<-ch
+	w1.Stop()
```

Comment:
maybe add a tiny sleep here to avoid flakiness? `<-ch` only hits that the workflow code hit that line, not that the worker completed the respondwftcompleted rpc